### PR TITLE
Image Grid Component

### PR DESCRIPTION
**Pull request by pablonajera**
## Ticket
[Trello](https://trello.com/c/l3ULoBgo)

# Description
Creating new `ImageGrid` component

## Props
| Name | Type | Required? | Description | Default Value |
| ----- | ---------- | ------- | -------- | -------- |
| `photos` | `array` | ✅ | An array of URLs for the images | |
| `numberOfColumns` | `number` | | Number of columns to show | defaults to being responsive based on window width |
| `spacing` | `number` | | Spacing between images (in pixels) | `10` |

### Screenshots
| Description | Prop values | Image |
| ----------- | ------ | ------ |
| Default full screen |  |  Screenshot 2023-09-20 at 12 53 50 AM |
| Default half screen | | Screenshot 2023-09-20 at 12 54 34 AM |
| 5 columns | `numberOfColumns`= 5 <br /><br />`spacing` = 10 | Screenshot 2023-09-20 at 12 52 45 AM |
| 2 Columns | `numberOfColumns`= 2 <br /><br /> `spacing` = 25 | Screenshot 2023-09-20 at 12 52 23 AM |

### Sample Usage
```jsx
const images = [
  "https://loremflickr.com/700/718",
  "https://loremflickr.com/520/821",
  "https://loremflickr.com/400/623",
  "https://loremflickr.com/420/901",
  "https://loremflickr.com/900/912",
  "https://loremflickr.com/300/281",
  "https://loremflickr.com/900/740",
  "https://loremflickr.com/780/791",
  "https://loremflickr.com/430/618",
  "https://loremflickr.com/900/841",
  "https://loremflickr.com/570/781",
  "https://loremflickr.com/800/918",
];

<ImageGrid
  numberOfColumns={5}
  spacing={10}
  photos={images}
></ImageGrid>
```

### CODE REVIEW

File: components/imageGrid/imageGrid.module.css [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.grid {
+  display: flex;
+  width: auto;
+}
+
+.column {
+  background-clip: padding-box;
```
Comment (angielamm) [inline components/imageGrid/imageGrid.module.css:7]: Im going to start using background clip!! love it

File: components/imageGrid/imageGrid.module.css [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+.column > img {
```
Comment (angielamm) [inline components/imageGrid/imageGrid.module.css:10]: Is this a conditional statement adjusting the img?

File: components/imageGrid/index.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import Masonry from 'react-masonry-css';
```
Comment (angielamm) [inline components/imageGrid/index.js]: What is *?

File: components/imageGrid/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  500: 1,
+};
+
+export default function ImageGrid({ photos, numberOfColumns, spacing }) {
+  const useStyles = createStyles(() => ({
+    columnSpacing: {
+      paddingLeft: `${spacing}px`,
+    },
+  }));
+  const { classes } = useStyles();
```
Comment (angielamm) [inline components/imageGrid/index.js:30]: little confused about this function, what's the difference between using this and imageGridStyles?

File: components/imageGrid/index.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+  const breakpointColumns = numberOfColumns
```
Comment (angielamm) [inline components/imageGrid/index.js:32]: when is the numberOfColumns condition met? when its a non-zero?

File: components/imageGrid/index.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    ? { default: numberOfColumns }
+    : defaultBreakpointColumns;
+
+  return (
+    <Masonry
+      className={imageGridStyles.grid}
+      breakpointCols={breakpointColumns}
+      columnClassName={cx(imageGridStyles.column, classes.columnSpacing)}
+    >
+      {photos.map((photo, index) => (
```
Comment (angielamm) [inline components/imageGrid/index.js:42]: I guess this function iterates over each photo and assigns it an index?

**Review by angielamm — DISMISSED** [review]
Looks awesome! It's much shorter than I anticipated the code would be, hopefully the integration for this one will be too lol

Love the KAT reference 😄

File: components/imageGrid/index.js (hunk 3 above)
Comment (pablonajera) [reply components/imageGrid/index.js]: ahh this is actually unnecessary from the previous PR.

File: components/imageGrid/imageGrid.module.css (hunk 2 above)
Comment (pablonajera) [reply components/imageGrid/imageGrid.module.css:10]: no this is a CSS inheritance selector, read more about it [here](https://www.geeksforgeeks.org/what-is-greater-than-sign-selector-in-css/)

File: components/imageGrid/index.js (hunk 4 above)
Comment (pablonajera) [reply components/imageGrid/index.js:30]: Good question! Doing it this way allows us to make dynamic styles based on JavaScript inputs. 

In other words, our styling will change depending on what the user passes in. There's no way to see our JS variables in our CSS file, so we can't do it in `imageGridStyles`

File: components/imageGrid/index.js (hunk 5 above)
Comment (pablonajera) [reply components/imageGrid/index.js:32]: Yes, non-zero exactly!

Comment (pablonajera) [reply components/imageGrid/index.js:32]: Read more about [js truthy here](https://developer.mozilla.org/en-US/docs/Glossary/Truthy)

File: components/imageGrid/index.js (hunk 6 above)
Comment (pablonajera) [reply components/imageGrid/index.js:42]: Yup! so we get the actual item as `photo` and the current index as `index`

File: components/imageGrid/imageGrid.module.css (hunk 2 above)
Comment (angielamm) [reply components/imageGrid/imageGrid.module.css:10]: Oh I see!! So this one adjusts all the imgs inside the column div?

**Comment by pablonajera** [comment]
@angielamm please re-approve 😅

**Review by angielamm — APPROVED** [review]

**Pull request merged**